url.bs

```diff
@@ -88,20 +88,30 @@ valid input. User agents, especially conformance checkers, are encouraged to rep
 
 <h3 id=parsers>Parsers</h3>
 
-<p>The <dfn>EOF code point</dfn> is a conceptual code point that signifies the end of a
-string or code point stream.
+<p>The <dfn>EOF code point</dfn> is a conceptual code point that signifies the end of a string or
+code point stream.
 
-<p>Within a parser algorithm that uses a <var>pointer</var> variable, <dfn>c</dfn>
-references the code point the <var>pointer</var> variable points to.
+<p>A <dfn>pointer</dfn> for a <a for=/>string</a> <var>input</var> is an integer that points to a
+<a for=/>code point</a> within <var>input</var>. Initially it points to the start of
+<var>input</var>. If it is &minus;1 it points nowhere. If it is greater than or equal to
+<var>input</var>'s <a for=string>code point length</a>, it points to the <a>EOF code point</a>.
 
-<p>Within a string-based parser algorithm that uses a <var>pointer</var> variable,
-<dfn>remaining</dfn> references the substring after <var>pointer</var> in the string
-being processed.
+<p>When a <a>pointer</a> is defined, <dfn>c</dfn> references the <a for=/>code point</a>
```

Comment:
What does "defined" mean? My first guess was "not -1", but reading the rest of the paragraph that doesn't seem right...

Maybe this is trying to say something like "each pointer has a c, which is a reference to the..."?

Reply:
Would "When a pointer is used" work?

Reply:
Yeah, that makes sense I think.